From: Ishant Vivek <ishantvivek1311@gmail.com>

Date: Mon, 6 June 2016 18:28:22 +0530

Subject: [PATCH] frameworks/base/core/jni/AndroidRuntime.cpp: ART to activate low memory

This is partially picked from low ram device configuration.

*ro.config_low_ram= true ,reduces acclgraphics too(reduces graphics),so this patch is modified to do only art optimization.

This is for low memory targets only(~512MB RAM & hdpi resolution),and here specified specially for bcom.

Also, set PRODUCT_PROPERTY_OVERRIDES += ro.optimize.bcm=true in the device makefile for this patch to take effect.

--- a/frameworks/base/core/jni/AndroidRuntime.cpp
+++ b/frameworks/base/core/jni/AndroidRuntime.cpp
@@ -699,6 +699,11 @@
     if (strcmp(propBuf, "true") == 0) {
       addOption("-XX:LowMemoryMode");
     }
+    
+property_get("ro.optimize.bcm", propBuf, "");
+    if (strcmp(propBuf, "true") == 0) {
+      addOption("-XX:LowMemoryMode");
+    }    
 
     parseRuntimeOption("dalvik.vm.gctype", gctypeOptsBuf, "-Xgc:");
     parseRuntimeOption("dalvik.vm.backgroundgctype", backgroundgcOptsBuf, "-XX:BackgroundGC=");
@@ -1501,4 +1506,4 @@
     return register_jni_procs(gRegJNI, NELEM(gRegJNI), env);
 }
 
-}   // namespace android
+}   // namespace android
\ No newline at end of file